app/controllers/locations_controller.rb

```diff
@@ -18,13 +19,17 @@ def external_search
     limit = location_params[:limit]
     if query.present?
       success, resp = Location.geosearch(query, limit)
-      if success
+      if success && resp.is_a?(Array)
+        # Successful request with results. Just keep the first if you get duplicate locations.
         results = resp.map { |r| LocationHelper.adapt_location_iq_response(r) }
-        # Just keep the first if you get duplicate locations
         results = results.uniq { |r| [r[:name], r[:geo_level]] }
+      elsif success && resp.is_a?(Hash) && resp["error"] == "Unable to geocode"
+        # Successful request but 0 results
+        Rails.logger.info("No geosearch results for: #{query}")
       else
```

Comment:
Is there no error message from the API? Nothing more specific we can log?

Reply:
the full response is actually `{ "error": "Unable to geocode" }`  :3

Reply:
I meant if there arent other errors... Can you guarantee that `success && resp.is_a?(Hash) && resp["error"] != "Unable to geocode"` does not happen?

Reply:
updated as discussed 👍